server/src/test/java/io/crate/auth/UserAuthenticationMethodTest.java

```diff
@@ -361,6 +375,136 @@ public void test_jwt_authentication_claim_mismatch_throws_error() throws Excepti
+            .build();
+        JWTAuthenticationMethod jwtAuth = new JWTAuthenticationMethod(
+            roles,
+            settings,
+            jwkProviderFunction(null),
+            () -> "test_cluster_id" // Default aud is not set, fallback to value from the token payload.
+        );
+
+        Credentials credentials = new Credentials(JWT_TOKEN);
+        credentials.setUsername(tokenUsername);
+
+        Role authenticatedRole = jwtAuth.authenticate(credentials, null);
+        assertThat(authenticatedRole.name()).isEqualTo(tokenUsername);
+    }
+
+    @Test
+    public void test_jwt_authentication_user_has_no_jwt_properties_default_is_not_set() throws Exception {
+        String tokenUsername = "cloud_user"; // Token payload dictates CrateDB username.
+        Roles roles = () -> List.of(
+            userOf(
+                tokenUsername,
+                Set.of(),
+                new HashSet<>(),
+                getSecureHash("pwd"),
+                // User doesn't have JWT properties
+                null
+            )
+        );
+        JWTAuthenticationMethod jwtAuth = new JWTAuthenticationMethod(
+            roles,
+            Settings.EMPTY,
+            jwkProviderFunction(null),
+            () -> "test_cluster_id" // Default aud is not set, fallback to value from the token payload.
+        );
+
+        Credentials credentials = new Credentials(JWT_TOKEN);
+        credentials.setUsername(tokenUsername);
+
+        // Auth fails because empty issuer doesn't match token's issuer.
+        assertThatThrownBy(
+            () -> jwtAuth.authenticate(credentials, null))
+            .isExactlyInstanceOf(RuntimeException.class)
+            .hasMessageContaining("jwt authentication failed for user cloud_user. Reason: The Claim 'iss' value doesn't match the required issuer.");
+    }
+
+    @Test
+    public void test_jwt_authentication_user_has_no_jwt_properties_default_values_dont_match_token_payload() throws Exception {
+        String tokenUsername = "cloud_user"; // Token payload dictates CrateDB username.
+        Roles roles = () -> List.of(
+            userOf(
+                tokenUsername,
+                Set.of(),
+                new HashSet<>(),
+                getSecureHash("pwd"),
+                // User doesn't have JWT properties
+                null
+            )
+        );
+        // Defaults doesn't match JWT_TOKEN payload.
+        var settings = Settings.builder()
+            .put(AuthSettings.AUTH_HOST_BASED_JWT_ISS_SETTING.getKey(), "dummy")
+            .put(AuthSettings.AUTH_HOST_BASED_JWT_AUD_SETTING.getKey(), "dummy")
+            .build();
+        JWTAuthenticationMethod jwtAuth = new JWTAuthenticationMethod(
+            roles,
+            settings,
+            jwkProviderFunction(null),
+            () -> "test_cluster_id" // Default aud is not set, fallback to value from the token payload.
```

Comment:
Is this comment really valuable and related to the test case?
```suggestion
            () -> "test_cluster_id" 
```